src/Bcgov/DesignSystemPlugin/DocumentRepository/src/Service/AdminUIManager.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Bcgov\DesignSystemPlugin\DocumentRepository\src\Service;
+
+use Bcgov\DesignSystemPlugin\DocumentRepository\src\Config\RepositoryConfig;
+
+/**
+ * AdminUIManager - Admin UI Integration
+ *
+ * This service handles all aspects of the WordPress admin UI integration,
+ * including menus, asset loading, and React app initialization.
+ */
+class AdminUIManager {
+    /**
+     * Configuration service.
+     *
+     * @var RepositoryConfig
+     */
+    private RepositoryConfig $config;
+
+    /**
+     * Document uploader service.
+     *
+     * @var DocumentUploader
+     */
+    private DocumentUploader $uploader;
+
+    /**
+     * Metadata manager service.
+     *
+     * @var DocumentMetadataManager
+     */
+    private DocumentMetadataManager $metadata_manager;
+
+    /**
+     * Constructor.
+     *
+     * @param RepositoryConfig        $config Configuration service.
+     * @param DocumentUploader        $uploader Document uploader service.
+     * @param DocumentMetadataManager $metadata_manager Metadata manager service.
+     */
+    public function __construct(
+        RepositoryConfig $config,
+        DocumentUploader $uploader,
+        DocumentMetadataManager $metadata_manager
+    ) {
+        $this->config           = $config;
+        $this->uploader         = $uploader;
+        $this->metadata_manager = $metadata_manager;
+    }
+
+    /**
+     * Add the document repository menu to the WordPress admin.
+     */
+    public function add_repository_menu(): void {
+        add_menu_page(
+            'Document Repository',
+            'Document Repository',
+            $this->config->get_capability(),
+            $this->config->get( 'menu_slug' ),
+            [ $this, 'render_repository_page' ],
+            $this->config->get( 'menu_icon' ),
+            $this->config->get( 'menu_position' )
+        );
+    }
+
+    /**
+     * Add the metadata settings submenu.
+     */
+    public function add_metadata_settings_submenu(): void {
+        add_submenu_page(
+            $this->config->get( 'menu_slug' ),
+            'Metadata Settings',
+            'Metadata Settings',
+            $this->config->get_capability(),
+            $this->config->get( 'metadata_slug' ),
+            [ $this, 'render_metadata_settings_page' ]
+        );
+    }
+
+    /**
+     * Enqueue admin scripts and styles.
```

Comment:
```suggestion
     * Enqueue admin scripts and styles for only the document repository application/metadata admin pages.
```
**Suggestion**: i think some additional clarity on the comment would be helpful to future self, and other devs